app/models/service_template_ansible_playbook.rb

```diff
@@ -179,6 +180,11 @@ def retirement_potential?
 
   private
 
+  def retain_resource_actions
+    @resource_actions = resource_actions
+    yield
+  end
```

Comment:
I feel we can implement the job template deletion after the `yield` thus remove the requirement for ab `after_destroy` call back. Also I am not sure we can still query `resource_actions` after the destroy. I am thinking of some thing like this
```
  job_templates = resource_actions.where.not(:configuration_template_id => nil).collect(&:configuration_template)
  yield
  job_templates.each do |jt|
    delete_in_provider
  end
```